model/task/task.go

```diff
@@ -3084,6 +3084,11 @@ func archiveAll(ctx context.Context, taskIds, execTaskIds, toRestartExecTaskIds
 
 	_, err = session.WithTransaction(ctx, txFunc)
 
+	// Return nil if the task has already been archived. Because we use a transaction here,
+	// we can trust that either the whole task has already been archived, or none of it.
```

Comment:
"we can trust that either all tasks have already been archived, or none of them"?

Reply:
true i was thinking the whole display task, but this is better